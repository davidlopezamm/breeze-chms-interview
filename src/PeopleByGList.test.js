import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';
import { MemoryRouter } from 'react-router';
import PeopleGroups from './pages/group/people_by_group';
import { Route } from 'react-router-dom';

configure({ adapter: new Adapter() });
let wrapper, data;

describe('<ResultsList /> People by group', () => {

    beforeAll(() => {
        wrapper = shallow(<PeopleGroups match={{params: {id: "Elders"}, isExact: true, path: "", url: ""}} />)

        data = [{
            "id": 132,
            "first_name": "David",
            "last_name": "Lopez",
            "email_address": "cremin.marjory@hotmail.com",
            "status": "active",
            "group_name": "Elders",
            "updated_at": "2019-07-20 22:05:47",
            "created_at": "2019-07-20 22:05:47"
        }]


        wrapper.setState({ 'data' : data });
    });

    test('should match the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
})
