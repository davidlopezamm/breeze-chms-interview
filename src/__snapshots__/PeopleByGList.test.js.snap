// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsList /> People by group should match the snapshot 1`] = `
<Table
  as="table"
  celled={true}
  sortable={true}
>
  <TableHeader
    as="thead"
  >
    <TableRow
      as="tr"
      cellAs="td"
    >
      <TableHeaderCell
        as="th"
        onClick={[Function]}
        sorted={null}
      >
        First Name
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        onClick={[Function]}
        sorted={null}
      >
        Last Name
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
        onClick={[Function]}
        sorted={null}
      >
        Email
      </TableHeaderCell>
      <TableHeaderCell
        as="th"
      >
        Action
      </TableHeaderCell>
    </TableRow>
  </TableHeader>
  <TableBody
    as="tbody"
  >
    <TableRow
      as="tr"
      cellAs="td"
      key="0"
    >
      <TableCell
        as="td"
        singleLine={true}
      >
        Macie
      </TableCell>
      <TableCell
        as="td"
        singleLine={true}
      >
        Emmerich
      </TableCell>
      <TableCell
        as="td"
        singleLine={true}
      >
        cremin.marjory@hotmail.com
      </TableCell>
      <TableCell
        as="td"
        singleLine={true}
      >
        <Link
          className="ui blue button"
          href="/people/form/132"
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Edit
        </Link>
        <Link
          className="ui red button"
          href="#"
          id={132}
          onClick={[Function]}
          onlyActiveOnIndex={false}
          style={Object {}}
        >
          Delete
        </Link>
      </TableCell>
      <Confirm
        cancelButton="Cancel"
        confirmButton="OK"
        content="Are you sure?"
        onCancel={[Function]}
        onConfirm={[Function]}
        open={false}
        size="small"
      />
    </TableRow>
  </TableBody>
</Table>
`;
